tools/nut-scanner/scan_xml_http.c

```diff
@@ -221,14 +257,22 @@ nutscan_device_t * nutscan_scan_xml_http(long usec_timeout)
 				(*nut_ne_xml_push_handler)(parser, startelm_cb,
 							NULL, NULL, nut_dev);
 				(*nut_ne_xml_parse)(parser, buf, recv_size);
+				int parserFailed = (*nut_ne_xml_failed)(parser); // 0 = ok, nonzero = fail
 				(*nut_ne_xml_destroy)(parser);
 
-				nut_dev->driver = strdup("netxml-ups");
-				sprintf(buf,"http://%s",string);
-				nut_dev->port = strdup(buf);
+				if (parserFailed == 0) {
+					nut_dev->driver = strdup("netxml-ups");
+					sprintf(buf,"http://%s",string);
+					nut_dev->port = strdup(buf);
 
-				current_nut_dev = nutscan_add_device_to_device(
+					current_nut_dev = nutscan_add_device_to_device(
 						current_nut_dev,nut_dev);
+				}
+				else
+				{
+					fprintf(stderr,"Device replied with NetXML but was not deemed compatible\n");
+					return NULL;
```

Comment:
the above "return" statement aborts the XML scan at the first discovered ePDU and breaks the upstream scan (not unitary) and MUST be removed. Otherwise, we end up with a big regression.
I'm also in favor of replacing the fprintf with another upsdebugx, since it's not an error in essence but a known limitation.